### ISSUE TITLE
can you use categorical data in dynamic features ?

### ISSUE THREAD

**Issue by Sandy4321** [brunoklein99/deepar#1]
can you use categorical data in dynamic features ?
vs 
paper
https://arxiv.org/pdf/1704.04110.pdf
One challenge often encountered when attempting to jointly learn from multiple time series in realworld forecasting problems is that the magnitudes of the time series differ widely, and the distribution of the magnitudes is strongly skewed.

**Comment by brunoklein99** [comment]
I don't understand your question. I don't see how your question relates to your citation. I will answer separately.

> can you use categorical data in dynamic features ?

If you have low cardinality categories they can be represented by one-hot-encoding, otherwise, by jointly trained embeddings.

>  One challenge often encountered when attempting to jointly learn from multiple time series in realworld forecasting problems is that the magnitudes of the time series differ widely, and the distribution of the magnitudes is strongly skewed.

You can attempt to normalize them for ease of convergence during training and handle the inverse operation as a post inference step. Although pure normalization won't completely solve skewness, you would need some kind of transformation.

**Comment by Sandy4321** [comment]
thanks for soon answer
If you have low cardinality categories they can be represented by one-hot-encoding, otherwise, by jointly trained embeddings
may you share code example of such a use?
you definitely may have it , since categorical data is often to meet in real live cases

**Comment by brunoklein99** [comment]
1. https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html
2. https://pytorch.org/docs/master/generated/torch.nn.Embedding.html
